packages/discord.js/src/structures/Channel.js

```diff
@@ -57,6 +57,15 @@ class Channel extends Base {
     return new Date(this.createdTimestamp);
   }
 
+  /**
+   * The URL to the channel
+   * @type {string}
+   * @readonly
+   */
+  get url() {
+    return `https://discord.com/channels/${this.isDMBased() ? '@me' : this.guild.id}/${this.id}`;
```

Comment:
This should use `guildId` since it's always guaranteed to be set:
```suggestion
    return `https://discord.com/channels/${this.isDMBased() ? '@me' : this.guildId}/${this.id}`;
```

Reply:
Good catch!